pfsimulator/parflow_lib/problem_van_genuchten.c

```diff
@@ -0,0 +1,658 @@
+//                       ipp = SubvectorEltIndex(pp_sub, i, j, k);
+//                       ipd = SubvectorEltIndex(pd_sub, i, j, k);
+                     {ind = SubvectorEltIndex(pd_alpha_sub, i, j, k);
+
+
+                       alpha = alphas[ir];
+                       n = ns[ir];
+                       s_res = s_ress[ir];
+                       s_sat = s_difs[ir] + s_ress[ir];  //new
+                       //s_dif = s_difs[ir]; // BB not needed
+
+                       pd_alpha_dat[ind] = alpha;  //BB
+                       pd_n_dat[ind] = n;  //BB
+                       pd_sres_dat[ind] = s_res;  //BB  // no ssat???
+                       pd_ssat_dat[ind] = s_sat;
+                     });
+      }     /* End subgrid loop */
+    }       /* End loop over regions */
+  }         /* End if data not from file */
+  else
+  {
+    gr_solid = ProblemDataGrDomain(problem_data);
+    n_values = dummy1->n_values;
+    alpha_values = dummy1->alpha_values;
+    s_res_values = dummy1->s_res_values;
+    s_sat_values = dummy1->s_sat_values;
+
+    ForSubgridI(sg, subgrids)
+    {
+      subgrid = SubgridArraySubgrid(subgrids, sg);
+//      ps_sub = VectorSubvector(phase_saturation, sg);
+//      pp_sub = VectorSubvector(phase_pressure, sg);
+//      pd_sub = VectorSubvector(phase_density, sg);
+
+      n_values_sub = VectorSubvector(n_values, sg);
+      alpha_values_sub = VectorSubvector(alpha_values, sg);
+      s_res_values_sub = VectorSubvector(s_res_values, sg);
+      s_sat_values_sub = VectorSubvector(s_sat_values, sg);
+
+      pd_alpha_sub = VectorSubvector(pd_alpha, sg);   //BB
+      pd_n_sub = VectorSubvector(pd_n, sg);           //BB
+      pd_sres_sub = VectorSubvector(pd_sres, sg);     //BB
+      pd_ssat_sub = VectorSubvector(pd_ssat, sg);     //BB
+
+      ix = SubgridIX(subgrid);
+      iy = SubgridIY(subgrid);
+      iz = SubgridIZ(subgrid);
+
+      nx = SubgridNX(subgrid);
+      ny = SubgridNY(subgrid);
+      nz = SubgridNZ(subgrid);
+
+      r = SubgridRX(subgrid);
+
+//      psdat = SubvectorData(ps_sub);
+//      ppdat = SubvectorData(pp_sub);
+//      pddat = SubvectorData(pd_sub);
+
+      n_values_dat = SubvectorData(n_values_sub);
+      alpha_values_dat = SubvectorData(alpha_values_sub);
+      s_res_values_dat = SubvectorData(s_res_values_sub);
+      s_sat_values_dat = SubvectorData(s_sat_values_sub);
+
+      pd_alpha_dat = SubvectorData(pd_alpha_sub);   //BB
+      pd_n_dat = SubvectorData(pd_n_sub);           //BB
+      pd_sres_dat = SubvectorData(pd_sres_sub);     //BB
+      pd_ssat_dat = SubvectorData(pd_ssat_sub);     //BB
+
+      GrGeomInLoop(i, j, k, gr_solid, r, ix, iy, iz, nx, ny, nz,
+                   {
+//                     ips = SubvectorEltIndex(ps_sub, i, j, k);
+//                   ipp = SubvectorEltIndex(pp_sub, i, j, k);
+//                   ipd = SubvectorEltIndex(pd_sub, i, j, k);
+                     ind = SubvectorEltIndex(pd_alpha_sub, i, j, k);
```

Comment:
The local variables in GrGeomInLoops should be defined locally also in this comp unit so that CUDA compilation can be enabled (although this comp unit does not use CUDA yet).